site/en/docs/extensions/mv3/messaging/index.md

```diff
@@ -289,57 +283,59 @@ Extensions [can exchange messages][native-client] with native applications that
 
 ### Content scripts are less trustworthy {: #content-scripts-are-less-trustworthy }
 
-[Content scripts are less trustworthy][security-cs] than the extension service worker (for example, a malicious web
-page might be able to compromise the renderer process where the content scripts run). Assume that
-messages from a content script might have been crafted by an attacker and make sure to [validate and
-sanitize all input][security-santize]. Assume any data sent to the content script might leak to the web page.
+[Content scripts are less trustworthy][security-cs] than the extension service worker.
+For example, a malicious webpage might be able to compromise the rendering process that
+runs the content scripts. Assume that messages from a content script might have been
+crafted by an attacker and make sure to [validate and sanitize all input][security-santize].
+Assume any data sent to the content script might leak to the web page.
 Limit the scope of privileged actions that can be triggered by messages received from content
 scripts.
 
 ### Cross-site scripting {: #cross-site-scripting }
 
-When receiving a message from a content script or another extension, your scripts should be careful
-not to fall victim to [cross-site scripting][wiki-cross-site]. This advice applies to scripts running inside the
-extension background page as well as to content scripts running inside other web origins.
-Specifically, avoid using dangerous APIs such as the ones below:
+When receiving a message from a content script or another extension, make sure to protect
+your scripts against [cross-site scripting][wiki-cross-site]. This advice applies to scripts running inside the
+extension background page as well as to content scripts running in other web origins. Use APIs that don't run
+scripts whenever possible:
 
 {% Label %}content-script.js or service-worker.js{% endLabel %}
 
 ```js
 chrome.tabs.sendMessage(tab.id, {greeting: "hello"}, function(response) {
-  // WARNING! Might be evaluating an evil script!
-  var resp = eval("(" + response.farewell + ")");
+  // JSON.parse doesn't evaluate the attacker's scripts.
+  var resp = JSON.parse(response.farewell);
```

Comment:
Please restore the old one. It was showing what not to do. [Use this widget](https://web.dev/handbook/web-dev-components/#compare-with-custom-labels) to label it as not good.

Reply:
@oliverdunk, can you please look over the script changes here.

Reply:
Did these get rearranged?

Reply:
I'm at least going to put the don't one after the do one, to reduce user confusion as much as possible. Readers often take the first example they find regardless of what flags we put on it.

Reply:
This looks ok to me.